coderedcms/models/page_models.py

```diff
@@ -1327,6 +1332,70 @@ def serve_submissions_list_view(self, request, *args, **kwargs):
         view = self.submissions_list_view_class.as_view()
         return view(request, form_page=self, *args, **kwargs)
 
+    def get_form(self, request, *args, **kwargs):
+        form_class = self.get_form_class()
+        form_params = self.get_form_parameters()
+        form_params.update(kwargs)
+
+        if request.method == 'POST':
+            return form_class(request.POST, request.FILES, *args, **form_params)
+        return form_class(*args, **form_params)
+
+    def contains_spam(self, request):
+        """
+        Checks to see if the spam honeypot was filled out.
+        """
+        if request.POST.get("cr-decoy-comments", None):
+            return True
+        return False
+
+    def process_spam_request(self, form, request):
+        """
+        Called when spam is found in the request.
+        """
+        messages.error(request, self.get_spam_message())
+        ip_address = utils.get_ip_from_request(request)
+        logger.info("{0} - Detected spam submission on page: {1}".format(ip_address, self.title))
```

Comment:
Should probably log the full request for the most helpful info. Plus if there is a false positive the submission would still be captured in the log. You wouldn't even need to format the IP address then because it would be contained in the request headers/info. I'm thinking something similar to the default django error message which spits out the request object (not sure how they do it, but you could probably steal that code).